app/controllers/items_controller.rb

```diff
@@ -75,9 +75,14 @@ def destroy
     
   def review
     @item = Item.find(params[:item_id])
-    @review = @item.reviews.build(user_id: current_user.id, body: params[:review][:body], rating: params[:review][:rating])
-    @review.save
-    redirect_to @item
+    if Review.find_by(user_id: current_user.id, reviewable: @item)
```

Comment:
You have access to both the User and the Item.

Rather than doing a strict find_by, it might be better to just directly check if a review exists by looking at the reviews belonging to the item, and then seeing if the given user created one of those reviews.

This chunk of code seems good either way though.